### DISCUSSION TITLE
Matching on ExitCode

### DISCUSSION THREAD

**Discussion by MichaelXavier** [chrisdone/hell#51] [Q&A] [answered]
I'm gathering from the API that `Process.runProcess` is meant to let you run processes that are allowed to fail. However, because there is no support yet for case statements and seemingly the constructors for ExitCode aren't exported and can't be constructed, it seems like there's no way to tell *whether* a process failed or not or check for particular exit codes. Am I missing anything?

**Comment by chrisdone** [comment, marked as answer]
That’s right, I haven’t added a deconstructor for it. Now that you point it out, seems that I’ve only written scripts that stop on the first failure. Happy to add an Exit.exitCode destructor.

It might be worth adding the other functions in https://hackage.haskell.org/package/base-4.20.0.1/docs/System-Exit.html
